app/lib/katello/resources/candlepin.rb

```diff
@@ -188,386 +152,26 @@ def upstream_consumer
-            cppath = join_path(path(id), "product/#{product_id}")
-            product = self.post(cppath, {}, self.default_headers)
-            JSON.parse(product).with_indifferent_access
-          end
-
-          def remove_product(id, product_id)
-            product = self.delete(join_path(path(id), "product/#{product_id}"), self.default_headers)
-            JSON.parse(product).with_indifferent_access
-          end
-
-          def add_pools(id, pool_id, quantity)
-            cppath = join_path(path(id), "pools/#{pool_id}")
-            quantity = Integer(quantity) rescue nil
-            cppath += "?quantity=#{quantity}" if quantity && quantity > 0
-            pool = self.post(cppath, {}, self.default_headers)
-            JSON.parse(pool).with_indifferent_access
-          end
-
-          def remove_pools(id, pool_id)
-            pool = self.delete(join_path(path(id), "pools/#{pool_id}"), self.default_headers)
-            JSON.parse(pool).with_indifferent_access
-          end
-
-          def content_overrides(id)
-            result = Candlepin::CandlepinResource.get(join_path(path(id), 'content_overrides'), self.default_headers).body
-            ::Katello::Util::Data.array_with_indifferent_access(JSON.parse(result))
-          end
-
-          # expected params
-          # id : ID of the Activation Key
-          # content_overrides => Array of content override hashes
-          def update_content_overrides(id, content_overrides)
-            attrs_to_delete = []
-            attrs_to_update = []
-            content_overrides.each do |content_override|
-              if content_override[:value]
-                attrs_to_update << content_override
-              else
-                attrs_to_delete << content_override
-              end
-            end
-
-            if attrs_to_update.present?
-              result = Candlepin::CandlepinResource.put(join_path(path(id), 'content_overrides'),
-                                                        attrs_to_update.to_json, self.default_headers)
-            end
-            if attrs_to_delete.present?
-              client = Candlepin::CandlepinResource.rest_client(Net::HTTP::Delete, :delete,
-                                                                join_path(path(id), 'content_overrides'))
-              client.options[:payload] = attrs_to_delete.to_json
-              result = client.delete({:accept => :json, :content_type => :json}.merge(User.cp_oauth_header))
-            end
-            ::Katello::Util::Data.array_with_indifferent_access(JSON.parse(result))
-          end
-
-          def path(id = nil)
-            "/candlepin/activation_keys/#{id}"
+      module PoolResource
+        def path(id = nil, owner_label = nil)
+          if owner_label && id
+            "#{prefix}/owners/#{owner_label}/pools/#{id}"
+          elsif owner_label
+            "#{prefix}/owners/#{owner_label}/pools/"
+          else
+            "#{prefix}/pools/#{id}"
```

Comment:
I think we can now remove [this line](https://github.com/Katello/katello/pull/7285/files#diff-d4ada19367f2e37f2adeafa9de778bccR182).

Reply:
Line 181 can still can be removed. I tested it locally, and we don't need it.